.github/workflows/angular.yml

```diff
@@ -191,6 +191,23 @@ jobs:
             -Dsonar.login=$SONAR_TOKEN
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+      - name: 'ANALYSIS: Local SonarQube PR Analysis'
+        if: >-
+          matrix.sonar-analyse == 'true' &&
+          steps.compare.outputs.equals != 'true' &&
+          matrix.workspaces != 'true' &&
+          matrix.skip-frontend-tests != 'true' &&
+          needs.build-matrix.outputs.client != 'false' &&
+          matrix.skip-backend-tests != 'true' &&
+          needs.build-matrix.outputs.server != 'false'
```

Comment:
simplify condition
```suggestion
        if: matrix.sonar-analyse == 'true'
```

Reply:
> if cache does not match, the overall issues should be shown.

If the cache does not match, it will fallback to an earlier cache, ensuring that we still have a baseline for comparison. Additionally, we should always have a valid cache after a PR is merged. While we could introduce a specific condition for this scenario, it would make the action significantly larger and more complex.

> simplify condition

Unfortunately, we cannot simplify this condition. SonarQube analysis requires test results, and as I understand, that's the reason these conditions exist in the original SonarQube step.

Reply:
> Unfortunately, we cannot simplify this condition. SonarQube analysis requires test results, and as I understand, that's the reason these conditions exist in the original SonarQube step.

Actually, we should force tests to make sure sonar is executed.
Adjusted the suggestion and we should adjust the build-matrix to run every test on frontend or backend changes when sonar-analyse is enabled.

Reply:
> Unfortunately, we cannot simplify this condition. SonarQube analysis requires test results, and as I understand, that's the reason these conditions exist in the original SonarQube step.

Actually, we should force tests to make sure sonar is executed.
Adjusted the suggestion and we should adjust the build-matrix to run every test on frontend or backend changes when sonar-analyse is enabled.

Reply:
I think we could just update the conditions on the steps themselves. But do we want to run the tests even if both `needs.build-matrix.outputs.server` and `needs.build-matrix.outputs.client` are false? Or should we in both steps check if either of them has changed?

Reply:
Which would be something like this:
```
      - name: 'TESTS: backend'
        id: backend
        if: >
          steps.compare.outputs.equals != 'true' && matrix.skip-backend-tests != 'true' &&
          (
            (matrix.sonar-analyse == 'true' && (needs.build-matrix.outputs.server != 'false' || needs.build-matrix.outputs.client != 'false')) ||
            needs.build-matrix.outputs.server != 'false'
          )
        run: npm run ci:backend:test
        continue-on-error: ${{matrix.continue-on-backend-tests-error || false}}
        timeout-minutes: 15
      - name: 'PREPARE: npm install'
        if: >
          steps.compare.outputs.equals != 'true' && matrix.skip-frontend-tests != 'true' &&
          (
            (matrix.sonar-analyse == 'true' && (needs.build-matrix.outputs.server != 'false' || needs.build-matrix.outputs.client != 'false')) ||
            needs.build-matrix.outputs.client != 'false'
          )
        run: ${{ (matrix.workspaces == 'true' && 'npm') || './npmw' }} install
        timeout-minutes: 7
      - name: 'TESTS: frontend'
        if: >
          steps.compare.outputs.equals != 'true' && matrix.skip-frontend-tests != 'true' &&
          (
            (matrix.sonar-analyse == 'true' && (needs.build-matrix.outputs.server != 'false' || needs.build-matrix.outputs.client != 'false')) ||
            needs.build-matrix.outputs.client != 'false'
          )
        run: npm run ci:frontend:test
        timeout-minutes: 15
```

Reply:
If `needs.build-matrix.outputs.server/client` are false, the workflow is not executed.

Reply:
Then is it safe to say `if: steps.compare.outputs.equals != 'true' && matrix.skip-backend-tests != 'true' && ((matrix.sonar-analyse == 'true'  || needs.build-matrix.outputs.server != 'false' )`?

Reply:
Looks ok.